isis/src/qisis/objs/JigsawSetupDialog/JigsawSetupDialog.cpp

```diff
@@ -1369,96 +1365,78 @@ namespace Isis {
 
-// Commented out since it contains some unimplemented functions (i.e. pseudo-code)
-
-  void JigsawSetupDialog::on_applySettingsPushButton_clicked() {}
-#if 0
-    // Get the current selected images
-    QAbstractProxyModel *model = m_ui->treeView->model();
-    // QItemSelectionModel *selectionModel = m_ui->treeView->selectionModel();
-    QModelIndexList selectedIndexes = m_ui->treeView->selectedIndexes();
+    QModelIndexList selectedIndexes = m_ui->treeView->selectionModel()->selectedIndexes();
     QStringList selectedObservationNumbers;
+
+    // Append selected images' serial numbers to selectedObservationNumbers
     foreach (QModelIndex index, selectedIndexes) {
-      QMap<int, QVariant> data = model->index(index);
-      qDebug() << "selected data: " << data;
-      // Cast the data to image (i think ProjectItem shows QVariant conversions)
-      // Get the observation number from the image
-      // Append the observation number to the selectedObservationNumbers
-    }
-
-    // check to see if all the images are selected
-
-    // If not, 
-    {  
-      // We are going to need to reset the bundle observation solve settings list for bundle settings
-      QList<BundleObservationSolveSettings> solveSettingsList;
-
-      // Grab the bundle settings
-      // (NOTE -- bundleSettings() needs to be changed so it can
-      // give us just a m_bundleSettings member we have for this current run of jigsaw, and the
-      // logic for filling the settings and bundle observation solve settings from the view needs
-      // to be put into two methods, like updateSettings() and updateBundleObservationSolveSettings())
-      BundleSettingsQsp bundleSettings = bundleSettings();
-
-      //find the bundle observation solve settings for each of the selected observations and remove
-      // the observation number from them 
-      for (int i = 0; i < bundleSettings->numberSolveSettings(); i++) {
-        foreach (observationNumber, selectedObservationNumbers) {
-          bundleSettings->observationSolveSettings(i).removeObservationNumber(observationNumber);
+      QModelIndex sourceIndex = proxyModel->mapToSource(index);
+      ProjectItem * projItem = sourceModel->itemFromIndex(sourceIndex);
+
+      if (projItem) {
+        // Tree traversal is top down so we dont need to do this check for imagelists?
+        if (projItem->isImage() && 
+            !selectedObservationNumbers.contains(projItem->image()->serialNumber())) {
+          selectedObservationNumbers.append(projItem->image()->serialNumber());
         }
-      }
-
-      // Remove any existing solve settings that no longer have any observation numbers
-      for (int i = 0; i < bundleSettings.numberSolveSettings(); i++) {
-        if (! bundleSettings->observationSolveSettings(i).observationNumbers().isEmpty() ) {
-          solveSettingsList.append(bundleSettings->observationSolveSettings(i));
+        else if (projItem->isImageList()) {
```

Comment:
Would it be possible to map back from the source selected image list item to the proxy image list item and its children? (I have no idea how feasible this is though.)

For example, outside of the *foreach* loop, map the source imagelist index to the proxy model, via (again, this may or may not work, and this code may be duplicated):

```
else if (projItem->isImageList()) {

  // Map the source image list item to the proxy model
  QModelIndex proxyIndex = proxyModel->mapFromSource(sourceIndex);

  // Find the children of the proxy model's image list
  foreach (auto &child : proxyModel->rowCount(proxyIndex)) {
    // Map the proxy image child back to the source
    QModelIndex sourceImageIndex = proxyModel->mapToSource(child->index());
    ProjectItem *sourceImage = sourceModel->itemFromIndex(sourceImageIndex);

    if (sourceImage->isImage() && !selectedObservationNumbers.contains(sourceImage->image()->serialNumber()) {
      selectedObservationNumbers.append(sourceImage->image()->serialNumber());
    }
  }